inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -59,62 +60,58 @@ bool MKLDNNPadNode::isSupportedOperation(const std::shared_ptr<const ngraph::Nod
 MKLDNNPadNode::MKLDNNPadNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
         : MKLDNNNode(op, eng, cache) {
     std::string errorMessage;
-    if (isSupportedOperation(op, errorMessage)) {
-        errorPrefix = "Pad node with name '" + op->get_friendly_name() + "'";
-        const auto pad = std::dynamic_pointer_cast<const ngraph::opset1::Pad>(op);
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    if (inputShapes.size() != 3 && inputShapes.size() != 4)
+        THROW_ERROR << " has incorrect number of input edges";
+    if (outputShapes.size() != 1)
+        THROW_ERROR << "Incorrect number of output edges";
+
+    const int srcDimsRank = inputShapes[DATA_ID].getRank();
```

Comment:
better to use `getInputShapeAtPort/getOutputShapeAtPort`

Reply:
Done, thanks